internal/internal_workflow_client.go

```diff
@@ -1661,6 +1661,120 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 	}, nil
 }
 
+func (w *workflowClientInterceptor) executeWorkflowWithOperation(
+	ctx context.Context,
+	startRequest *workflowservice.StartWorkflowExecutionRequest,
+	operation WithStartWorkflowOperation,
+) (*workflowservice.StartWorkflowExecutionResponse, error) {
+	startOp := &workflowservice.ExecuteMultiOperationRequest_Operation{
+		Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+			StartWorkflow: startRequest,
+		},
+	}
+
+	var withStartOp *workflowservice.ExecuteMultiOperationRequest_Operation
+	switch t := operation.(type) {
+	case *UpdateWorkflowOperation:
+		if err := t.markExecuted(); err != nil {
+			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
+		}
+
+		updateReq, err := w.createUpdateWorkflowRequest(ctx, t.input)
```

Comment:
What header does the update request end up getting when using the tracing interceptor? I wonder how this multi op will display in traces? @cretz I think Python modled signal with start a similar way how do the traces appear there?

Reply:
Yeah signal with start is a single call with a single header that I'm guessing becomes separate headers on the separate start/signal events worker side. Here I would expect we'd populate both the start and update headers with the inbound headers and they would remain/appear separate in the separate events worker side.

Reply:
I see there are two tracing methodologies: OpenTelemetry and OpenTracing. Would this be needed for both or just one (given OpenTracing's standing compared to OTEL)?

Reply:
Also (I could be wrong), I don't see a tracing integration for Update itself anywhere.

Reply:
There is a common tracing interceptor they all use, it's just a different tracer they provide (they all ultimately call https://pkg.go.dev/go.temporal.io/sdk/interceptor#NewTracingInterceptor).

Upon review, you are correct, we didn't properly wire `ClientOutboundInterceptor.UpdateWorkflow` in the tracing interceptor to populate header or `WorkflowInboundInterceptor.ExecuteUpdate` to rehydrate span from header. We need to fix that, it does work in other SDKs. This can be fixed separately (I have opened https://github.com/temporalio/sdk-go/issues/1588). It does look like we did context propagation though (a prereq).

I do still think we need to make sure that context propagation works (i.e. that `headerPropagated` is being called outbound) which seems to be the case so we're probably good here until that other issue is tackled. If you'd like, you can add an integration test like `TestContextPropagator` for multi-op.